### PR TITLE
Use go-enry for language detection

### LINKED ISSUE #115: search_symbols fails with None is not iterable
Hello! First of all, great work you are doing, I really love it!

I've tried your new search_symbols feature and it seems to work in general, but apparently not in all cases and it is not clear why.

The content of the response is None such that json() returns None which is not iterable so it fails: https://github.com/hide-org/hide-sdk/blob/main/hide/client/hide_client.py#L197 

https://github.com/django/django/ is one repository where I see that it fails. I assume this is related to some problem with the LSP. I see that warning, but it probably is not my setup. Other Python repositories work fine for me. Any idea what might cause this?

### PR DESCRIPTION
**Pull request by artmoskvin**
This PR fixes #115 where the project language was detected incorrectly. The previous language detection algorithm naively counted file extensions and returned the language corresponding to the most popular file extension. 

The new algorithm is based on https://github.com/go-enry/go-enry, SOTA for language detection. The language of the file is detected by `enry` and the project language is defined as the language having most bytes, not just files. This algorithm is still not ideal but it's good enough and corresponds to how GitHub detects languages for the repos. 

I tested it on the reported repo `django` and it was accurate.

### CODE REVIEW

File: cmd/run.go [hunk 1]
```diff
@@ -95,13 +95,13 @@ var runCmd = &cobra.Command{
 		projectsDir := filepath.Join(home, HidePath, ProjectsDir)
 
 		lspServerExecutables := make(map[lsp.LanguageId]lsp.Command)
-		lspServerExecutables[lsp.LanguageId("go")] = lsp.NewCommand("gopls", []string{})
-		lspServerExecutables[lsp.LanguageId("python")] = lsp.NewCommand("pyright-langserver", []string{"--stdio"})
-		lspServerExecutables[lsp.LanguageId("javascript")] = lsp.NewCommand("typescript-language-server", []string{"--stdio"})
-		lspServerExecutables[lsp.LanguageId("typescript")] = lsp.NewCommand("typescript-language-server", []string{"--stdio"})
+		lspServerExecutables[lsp.Go] = lsp.NewCommand("gopls", []string{})
```
Comment (aleh-null) [inline cmd/run.go:98]: I think we should move all this logic to `lsp` pkg. Something like `lsp.Excecutables() map[lsp.LanguageId]lsp.Command`

**Review by aleh-null — COMMENTED** [review]
Really nice! I wonder if we really need to pinpoint main language in a repo, I like the way do count for each language.

**Comment by artmoskvin** [comment]
> I wonder if we really need to pinpoint main language in a repo, I like the way do count for each language.

we don't really pin it. we just need to know which lsp server to start.

File: cmd/run.go (hunk 1 above)
Comment (artmoskvin) [reply cmd/run.go]: done

**Review by aleh-null — APPROVED** [review]

**Pull request merged**